net/socket/accept.c

```diff
@@ -259,29 +260,36 @@ int accept(int sockfd, FAR struct sockaddr *addr, FAR socklen_t *addrlen)
       goto errout;
     }
 
-  /* Allocate a socket descriptor for the new connection now (so that it
-   * cannot fail later)
-   */
-
-  newfd = sockfd_allocate(&newsock, O_RDWR);
-  if (newfd < 0)
+  newsock = kmm_zalloc(sizeof(*newsock));
+  if (newsock == NULL)
     {
-      errcode = ENFILE;
+      errcode = ENOMEM;
       goto errout;
     }
 
   ret = psock_accept(psock, addr, addrlen, newsock);
   if (ret < 0)
     {
       errcode = -ret;
-      goto errout_with_socket;
+      goto errout_with_alloc;
+    }
+
+  /* Allocate a socket descriptor for the new connection now (so that it
+   * cannot fail later)
+   */
+
+  newfd = sockfd_allocate(newsock, O_RDWR);
+  if (newfd < 0)
+    {
+      errcode = ENFILE;
+      goto errout_with_alloc;
     }
 
   leave_cancellation_point();
   return newfd;
 
-errout_with_socket:
-  nx_close(newfd);
+errout_with_alloc:
+  kmm_free(newsock);
```

Comment:
i suspect you should use psock_close here.

Reply:
Done